### PR TITLE
Lint support

### PR DESCRIPTION
**Pull request by tatsuhama**
add Android Lint

### CODE REVIEW

File: test/lint.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+$LOAD_PATH.unshift File.expand_path(File.join(File.dirname($0), "..", "lib"))
+require "dokumi"
+
+class TestFindbugs < Minitest::Test
+  def build_script
+    File.expand_path(File.join(File.dirname($0), "build", "dokumi-test-android-lint.rb"))
+  end
+
+  def test_static_analysis
+    issues = Dokumi::Command.review("github.com", "tatsuhama", "DokumiLint", 1, skip_comment_creation: true, build_script: build_script)
```
Comment (giginet) [inline test/lint.rb]: Could you also send PR to cookpad/dokumi-test and use it?

https://github.com/cookpad/dokumi-test/pulls

File: test/lint.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    
+    assert_equal 2, issues.length
+    issues.each do |issue|
+      assert_equal :lint, issue[:tool]
```
Comment (giginet) [inline test/lint.rb]: According to our styleguide you should use `{}` syntax on the statement in one line.
https://github.com/cookpad/styleguide/blob/master/ruby.en.md#method-calls

`issues.each { |issue| assert_equal :lint, issue[:tool] }`

File: lib/dokumi/tool/android/lint.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          LINT_REPORT_FILE = "build/outputs/lint-results-debug.xml"
+        
+        class << self
+          def parse_report(target_project)
+            report_path = Support.make_pathname(target_project).join(LINT_REPORT_FILE)
+            File.open(report_path) do |file|
+              report = Nokogiri::XML(file)
+
+              report.xpath("//issue").map do |issue|
+                {
```
Comment (giginet) [inline lib/dokumi/tool/android/lint.rb]: According to our styleguide, you should not break lines here.

https://github.com/cookpad/styleguide/blob/master/ruby.en.md#hashes

**Review by giginet — CHANGES_REQUESTED** [review]
Thank you for your contribution 😄 

I commented some issues.

**Comment by tatsuhama** [comment]
@giginet 
I fixed your comments.
Please check my commits.

File: lib/dokumi/tool/android/lint.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      class Lint
+          LINT_REPORT_FILE = "build/outputs/lint-results.xml"
+        
+        class << self
+          def parse_report(target_project)
+            report_path = Support.make_pathname(target_project).join(LINT_REPORT_FILE)
+            File.open(report_path) do |file|
+              report = Nokogiri::XML(file)
+
+              report.xpath("//issue").map do |issue| {
```
Comment (giginet) [inline lib/dokumi/tool/android/lint.rb]: I'm sorry confusing you 🙇 , linebreak is needed here.
I meant

``` ruby
report.xpath("//issue").map do |issue|
  { description: issue.attribute("message").to_s,
    file_path: issue.xpath("location/@file").to_s,
    line: issue.xpath("location/@line").to_s.to_i,
    type: issue.attribute("severity").to_s == "Error" ? :error : :warning,
  }
end
```

opening bracket of hashes requires to be same line with first element.

**Review by giginet — CHANGES_REQUESTED** [review]

File: lib/dokumi/tool/android/lint.rb (hunk 4 above)
Comment (tatsuhama) [reply lib/dokumi/tool/android/lint.rb]: @giginet 
I fixed.
Please check my commit.
https://github.com/cookpad/dokumi/pull/12/commits/68746516b9c4930f55840318683198e9f06a7c92

**Review by giginet — APPROVED** [review]

**Comment by giginet** [comment]
I'm sorry. I made you wait.
I asked an Android engineer to check this PR, therefore it seems to LGTM.
I'll merge.

Thank you contribution 🎉

**Pull request merged**